client/subscription-product-onboarding/toast.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * External dependencies
+ */
+import { useDispatch } from '@wordpress/data';
+import { useEffect } from '@wordpress/element';
+import { registerPlugin } from '@wordpress/plugins';
+import { removeQueryArgs } from '@wordpress/url';
+import { __ } from '@wordpress/i18n';
+
+const SubscriptionProductOnboardingToast = () => {
+	const { createInfoNotice } = useDispatch( 'core/notices' );
+
+	useEffect( () => {
+		if ( window?.history ) {
+			window.history.replaceState(
+				null,
+				null,
+				removeQueryArgs(
+					window.location.href,
+					'wcpay-subscriptions-onboarded'
+				)
+			);
+		}
+
+		createInfoNotice(
+			__(
+				'Thank you for setting up WooCommerce Payments! We’ve published your first subscription product.',
+				'woocommerce-payments'
+			)
+		);
+	}, [ createInfoNotice ] );
```

Comment:
Yeah, admittedly `useEffect ` is a bit confusing 😀 You are correct though. The second param is an array of dependencies for the effect. If any of those dependencies change then the effect will run again.

It doesn't make much sense in this scenario as `createInfoNotice` will not change between re-renders. But you must list all dependencies that are used inside the effect otherwise eslint complains.

Additional reading:
- https://reactjs.org/docs/hooks-rules.html
- https://github.com/facebook/react/issues/14920

Reply:
Oh ok. That makes sense. Thanks for explaining. :)